pkg/result/ignore.go

```diff
@@ -83,19 +92,26 @@ func (i *IgnoreFinding) UnmarshalYAML(value *yaml.Node) error {
 
 type IgnoreFindings []IgnoreFinding
 
-func (f *IgnoreFindings) Match(id, path string, pkg *packageurl.PackageURL) *IgnoreFinding {
+func (f *IgnoreFindings) Match(id string, results *FindingsResults) *IgnoreFinding {
 	for _, finding := range *f {
 		if id != finding.ID {
 			continue
 		}
-		if !matchPath(path, finding.Paths) || !matchPURL(pkg, finding.PURLs) {
+		if !matchPath(results.TargetPath, finding.Paths) || !matchPURL(results.PackageURL, finding.PURLs) {
 			continue
 		}
 
-		log.Debug("Ignored", log.String("id", id), log.String("target", path))
-		return &finding
+		if finding.StartLine != 0 && finding.EndLine != 0 &&
```

Comment:
Should only those finds that exactly match the start and end of the lines be ignored? Or is it better to ignore everything between the passed position? For example, I want to ignore a misconfig that is found inside a Terraform block, but I want to specify the position of the block.

Reply:
I went back and forth here and thought it made more sense to be explicit to match the lines of the finding itself instead of a range.

Reply:
@michael-a-shelton could you give an example of why passing a range isn't ideal? Just trying to understand. To me it feels passing an explicit range would have more flexibility.

Reply:
For instance, someone typos the "end_line" in their ignore file. Maybe they meant to do:
```yaml
start_line: 10
end_line: 15
```

but instead, fat-fingered the number on end_line so it was something like:
```yaml
start_line: 10
end_line: 100
```

This would still pass if using a range, but should actually instead fail as now you're ignoring more than intended.

The start and end line numbers are actually how the finding itself is captured as well, so those fields are there mirroring the CauseMetadata struct for the misconf: https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/types/misconf.go#L37-L38